signal/src/lib.rs

```diff
@@ -558,11 +564,12 @@ pub mod pallet {
 
 			ProposalVoting::<T>::insert(&proposal_id, &voting);
 
-			// SBP-M2 review: what if proposal is not finalized?
+			// ? | SBP-M2 review: what if proposal is not finalized?
 			// Is it properly handled?
+			// yura: how should we handle None proposal_state?
```

Comment:
Logging appropriate message, containing `proposal_id` should work here.

Reply:
Added logs.